lib/cookie.c

```diff
@@ -264,6 +264,11 @@ static const char *get_top_domain(const char * const domain, size_t *outlen)
   return first? first: domain;
 }
 
+/* Avoid C1001.  See https://github.com/curl/curl/issues/6112 */
```

Comment:
I would like this to be a self-containend description of the issue at hand and not just a link. The link can be there as well, but it shouldn't be required for understanding why the `pragma` is there.

Reply:
@plujon  ?

Reply:
Feel free to suggest a description, or modify my change directly if you like.  C1001 is a code for an ICE (internal compiler error).  I'm not sure what you'd like said about it.  It's a bug in some older versions of Microsoft's compiler.

Reply:
I can pick this up and write up a description and push it.